### PR TITLE
Generate React Native code

### PR DESCRIPTION
**Pull request by roeierez**
This PR is the starting of an effort to fully generate the Breez SDK React Native package code.
It uses uniffi plugin model.
To use this utility one should supply the breez-sdk-binding directory and the output which is the root of the react native package.

For example

`cargo run -- -b breez-sdk/libs/sdk-bindings -o breez-sdk/libs/sdk-react-native/`

In order to check that the code compiles without errors and warnings we can run kotlin compiler without the need to build a sample app, for example:

`kotlinc BreezSDKMapper.kt breez_sdk.kt -cp ./classes.jar:./jna-5.12.1.jar -d .`

Notice that the compilation depends on two jar files that needs to exist on the classpath:
`jna-5.12.1.jar` - The jar needed for the java native interface and can be downloaded.
`classes.jar` - This jar is the React Native classes that can be found under the react native node module.

The plan is to add the compilation as part of the class generation later on.

- [x] Generate BreezSDKMapper.kt
- [x] Generate BreezSDKMapper.swift
- [ ] Add complication verification as part of the generation process
- [x] Generate BreezSDKModule.kt
- [x] Generate BreezSDKModule.swift
- [x] Generate javascript code
- [ ] Generate kotlin callbacks and listeners
- [ ] Generate Swift callbacks and listeners

### CODE REVIEW

**Comment by dangeross** [comment]
Looking good! I was able to run the generator no problem, but not able to test it. I couldn't find a classes.jar file usable locally to run the kotlinc, so it wasn't able to resolve `com.facebook.react.bridge`

Some thoughts:
1. Seems the output variable names are snake case, ideally they are camel case in kotlin.
2. Maybe consider grouping the kotlin generator into a kotlin module

**Review by dangeross — APPROVED** [review]
LGTM! I made some comments that could be addressed later. Is the plan to create a PR for each list item?

File: src/templates/Helpers.kt
Comment (dangeross) [reply src/templates/Helpers.kt]: I committed a change to track the sequence types used in structs and generate the pushToArray function from that. Still needed is to additionally track the function return types when they are also a sequence.

Comment (roeierez) [reply src/templates/Helpers.kt]: Looks good!

File: src/templates/Helpers.kt
Comment (roeierez) [reply src/templates/Helpers.kt]: Removed

**Comment by roeierez** [comment]
> 2\. Maybe consider grouping the kotlin generator into a kotlin module

Sure, done. Now we can duplicate the module and mimic the same for swift.

**Review by dangeross — APPROVED** [review]
Great! Generates and compiles with the provided jar files.

**Comment by dangeross** [comment]
@roeierez I would suggest we merge this as is, if you want to do a final review. 

I've adapted the event/log listeners in breez/breez-sdk#429 to emit with data from the generated mappers. They can then be improved alongside the per-instance log handling and adapted to handle multiple instances.

**Pull request merged**
